docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,373 @@
+****[//]: # (title: Compatibility guide for Kotlin 1.7)
+
+_[Keeping the Language Modern](kotlin-evolution.md)_ and _[Comfortable Updates](kotlin-evolution.md)_ are among the fundamental principles in
+Kotlin Language Design. The former says that constructs which obstruct language evolution should be removed, and the
+latter says that this removal should be well-communicated beforehand to make code migration as smooth as possible.
+
+While most of the language changes were already announced through other channels, like update changelogs or compiler
+warnings, this document summarizes them all, providing a complete reference for migration from Kotlin 1.6 to Kotlin 1.7.
+
+## Basic terms
+
+In this document we introduce several kinds of compatibility:
+
+- _source_: source-incompatible change stops code that used to compile fine (without errors or warnings) from compiling
+  anymore
+- _binary_: two binary artifacts are said to be binary-compatible if interchanging them doesn't lead to loading or
+  linkage errors
+- _behavioral_: a change is said to be behavioral-incompatible if the same program demonstrates different behavior
+  before and after applying the change
+
+Remember that those definitions are given only for pure Kotlin. Compatibility of Kotlin code from the other languages
+perspective
+(for example, from Java) is out of the scope of this document.
+
+## Language
+
+<!--
+### Title
+
+> **Issue**: [KT-NNNNN](https://youtrack.jetbrains.com/issue/KT-NNNNN)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**:
+>
+> **Deprecation cycle**:
+>
+> - 1.5.20: warning
+> - 1.7.0: report an error
+-->
```

Comment:
Do you want to keep this template?

Reply:
Yes, it would be very handy for further updates and next version compatibility guides